src/app-profile.js

```diff
@@ -0,0 +1,391 @@
+ * });
+ */
+AppProfile.prototype.get = function(gaxOptions, callback) {
+  var self = this;
+
+  if (is.fn(gaxOptions)) {
+    callback = gaxOptions;
+    gaxOptions = {};
+  }
+
+  this.getMetadata(gaxOptions, function(err, metadata) {
+    callback(err, err ? null : self, metadata);
+  });
+};
+
+/**
+ * Get the app profile metadata.
+ *
+ * @param {object} [gaxOptions] Request configuration options, outlined
+ *     here: https://googleapis.github.io/gax-nodejs/CallSettings.html.
+ * @param {function} callback The callback function.
+ * @param {?error} callback.err An error returned while making this
+ *     request.
+ * @param {object} callback.metadata The metadata.
+ * @param {object} callback.apiResponse The full API response.
+ *
+ * @example
+ * appProfile.getMetadata(function(err, metadata, apiResponse) {});
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * appProfile.getMetadata().then(function(data) {
+ *   var metadata = data[0];
+ *   var apiResponse = data[1];
+ * });
+ */
+AppProfile.prototype.getMetadata = function(gaxOptions, callback) {
+  var self = this;
+
+  if (is.fn(gaxOptions)) {
+    callback = gaxOptions;
+    gaxOptions = {};
+  }
+
+  this.bigtable.request(
+    {
+      client: 'BigtableInstanceAdminClient',
+      method: 'getAppProfile',
+      reqOpts: {
+        name: this.id,
+      },
+      gaxOpts: gaxOptions,
+    },
+    function() {
+      if (arguments[1]) {
+        self.metadata = arguments[1];
+      }
+
+      callback.apply(null, arguments);
+    }
+  );
+};
+
+/**
+ * Set the app profile metadata.
+ *
+ * @param {object} metadata See {@link Instance#createAppProfile} for the
+ *     available metadata options.
+ * @param {object} [gaxOptions] Request configuration options, outlined here:
+ *     https://googleapis.github.io/gax-nodejs/CallSettings.html.
+ * @param {function} callback The callback function.
+ * @param {?error} callback.err An error returned while making this request.
+ * @param {object} callback.apiResponse The full API response.
+ *
+ * @example
+ * const Bigtable = require('@google-cloud/bigtable');
+ * const bigtable = new Bigtable();
+ * const instance = bigtable.instance('my-instance');
+ * const appProfile = instance.appProfile('my-appProfile');
+ *
+ * const metadata = {
+ *   description: 'My Updated App Profile',
+ *   routing: myCluster,
+ *   allowTransactionalWrites: true,
+ * };
+ *
+ * appProfile.setMetadata(metadata, callback);
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * appProfile.setMetadata(metadata).then(function(data) {
+ *   const apiResponse = data[0];
+ * });
+ */
+AppProfile.prototype.setMetadata = function(metadata, gaxOptions, callback) {
+  if (is.fn(gaxOptions)) {
+    callback = gaxOptions;
+    gaxOptions = {};
+  }
+
+  var reqOpts = {
+    appProfile: AppProfile.formatAppProfile_(metadata),
+    updateMask: {
+      paths: [],
+    },
+  };
+  reqOpts.appProfile.name = this.id;
```

Comment:
This is what is required when I was trying to get updates working. Putting the name as a property of `reqOpts` didn't do anything